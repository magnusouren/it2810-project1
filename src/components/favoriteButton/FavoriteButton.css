/* Favorite Toggle */
.favorite-toggle {
  font-size: 2em;
  color: #fbc4ab;
  margin-right: 0.5em;
  font-weight: 300;
  position: absolute;
  top: 36px;
  right: 30px;
  cursor: pointer;
  transition: 0.02s ease-in;
  user-select: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-touch-callout: none;
}

.favorite-toggle:hover {
  transform: scale(1.1);
}

.favorite-toggle:active {
  transform: scale(1);
}

.favorite {
  color: #f00;
}

@media screen and (width <= 992px) {
  .favorite-toggle {
    top: 90px;
  }
}
