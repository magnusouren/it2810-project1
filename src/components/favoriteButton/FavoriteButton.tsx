import './FavoriteButton.css';

import { FC, useState } from 'react';
import { FaHeart, FaRegHeart } from 'react-icons/fa';

import { isFavorite, toggleFavorite } from '../../utils/persistency';

interface FavoriteButtonProps {
  id: string;
}

export const FavoriteButton: FC<FavoriteButtonProps> = ({ id }) => {
  const [isFavoriteDrink, setIsFavoriteDrink] = useState(isFavorite(id));

  const updateFavoriteDrink = () => {
    setIsFavoriteDrink(id ? isFavorite(id) : false);
  };

  const toggleFavoriteDrink = () => {
    if (id) {
      toggleFavorite(id);
      updateFavoriteDrink();
    }
  };

  const heart = isFavoriteDrink ? <FaHeart /> : <FaRegHeart />;

  return (
    <div
      className={`favorite-toggle${isFavoriteDrink ? ' favorite' : ''}`}
      onClick={toggleFavoriteDrink}
      data-testid='favorite'
    >
      {heart}
    </div>
  );
};
