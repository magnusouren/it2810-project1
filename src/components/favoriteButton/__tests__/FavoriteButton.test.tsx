import { act, cleanup, render, screen } from '@testing-library/react';
import { afterAll } from 'vitest';

import { FavoriteButton } from '../FavoriteButton';

describe('FavoriteButton', () => {
  afterAll(cleanup);
  it('Should match snapshot', () => {
    const { container } = render(<FavoriteButton id='1' />);
    expect(container).toMatchSnapshot();
  });

  it('Toggle should change class after click', () => {
    render(<FavoriteButton id='1' />);
    act(() => {
      screen.getByText('favorite').parentElement?.click();
    });
    expect(screen.getByText('favorite').parentElement?.classList.contains('favorite')).toBe(true);
    act(() => {
      screen.getByText('favorite').parentElement?.click();
    });
    expect(screen.getByText('favorite').parentElement?.classList.contains('favorite')).toBe(false);
  });
});
