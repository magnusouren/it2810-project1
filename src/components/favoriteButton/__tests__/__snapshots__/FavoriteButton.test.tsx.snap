// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FavoriteButton > Should match snapshot 1`] = `
<div>
  <div
    class="favorite-toggle"
  >
    <span
      class="material-symbols-outlined"
    >
      favorite
    </span>
  </div>
</div>
`;
